ntoskrnl/ke/thrdschd.c

```diff
@@ -74,13 +74,84 @@ KiQueueReadyThread(IN PKTHREAD Thread,
     KxQueueReadyThread(Thread, Prcb);
 }
 
+#ifdef CONFIG_SMP
+ULONG
+NTAPI
+KiFindIdealProcessor(
+    _In_ KAFFINITY ProcessorSet,
+    _In_ UCHAR OriginalIdealProcessor)
```

Comment:
I wonder whether you could go further and do:
```c
VOID
NTAPI
KiFindIdealProcessor(
    _In_ KAFFINITY ProcessorSet,
    _Inout_ PUCHAR IdealProcessor)
```
where in input, `*IdealProcessor` is this original ideal processor, and in output, it gets updated if needed.